packages/astro-og-canvas/src/routing.ts

```diff
@@ -4,8 +4,7 @@ import type { OGImageOptions } from './types';
 
 const pathToSlug = (path: string): string => {
   path = path.replace(/^\/src\/pages\//, '');
-  const extIndex = path.lastIndexOf('.');
-  path = path.slice(0, extIndex) + '.png';
+  path = path.replace(/\..*$/, '') + '.png';
```

Comment:
Sorry to take a while to get to this, going to get this merged now!

Making one change here though:

```suggestion
  path = path.replace(/\.[^\.]*$/, '') + '.png';
```

This makes sure we don’t truncate too aggressively in cases where the path is something like `'thing.something.md'`. `replace(/\..*$/, '')` would give `'thing'`, instead of `'thing.something'`.

Reply:
Good catch. Thanks for the heads-up!